### PR TITLE
Add section to sanitizer doc for `-Zexternal-clangrt`

### PR DESCRIPTION
**Pull request by ObsidianMinor**
After spending a week looking for answers to how to do the very thing this flag lets me do, it felt appropriate to document it where I would've expected it to be.

### CODE REVIEW

**Comment by ehuss** [comment]
r? michaelwoerister

It's not clear to me why this wasn't documented.

**Comment by ObsidianMinor** [comment]
cc @chriswailes as well

**Comment by rcvalle** [comment]
LGTM.

**Comment by chriswailes** [comment]
Thanks for taking care of this. Sorry I didn't add the documentation when I landed the patch.

**Comment by ObsidianMinor** [comment]
Wanted to add a small tidbit near the top with all the other flags to make sure it can't be missed.

File: src/doc/unstable-book/src/compiler-flags/sanitizer.md [hunk 1]
```diff
@@ -45,6 +45,8 @@ To enable a sanitizer compile with `-Zsanitizer=address`, `-Zsanitizer=cfi`,
 `-Zsanitizer=dataflow`,`-Zsanitizer=hwaddress`, `-Zsanitizer=leak`,
 `-Zsanitizer=memory`, `-Zsanitizer=memtag`, `-Zsanitizer=shadow-call-stack`, or
 `-Zsanitizer=thread`. You might also need the `--target` and `build-std` flags.
+If you're working with other languages that are also instrumented with sanitizers,
+you might need the `external-clangrt` flag. See the section on [working with other languages](#working-with-other-languages).
```
Comment (rcvalle) [inline src/doc/unstable-book/src/compiler-flags/sanitizer.md:49]: Maybe wrap before 80 chars?

**Comment by michaelwoerister** [comment]
Thanks for the PR, @ObsidianMinor! I'm sorry, you wasted time due to the missing documentation.

Would you mind also adding an `external-clangrt.md` file to the `compiler-flags` directory for increased visibility? It can refer to `sanitizer.md` for the more detailed explanation.

**Comment by michaelwoerister** [comment]
Looks great -- thanks, @ObsidianMinor!

@bors r+ rollup

**Pull request merged**